src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPageCount, setPerPageCount] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const totalItems = 42;
+
+  const firstItemOnPage = (currentPage - 1) * perPageCount;
+
+  const lastItemOnPage = firstItemOnPage + perPageCount;
+
+  const amountOfItems = getNumbers(1, 42).filter(
+    a => a >= firstItemOnPage + 1 && a <= lastItemOnPage,
+  );
+
+  const visibleItems = amountOfItems.map(n => `Item ${n}`);
+
+  const handleChangeOption = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPageCount(+event.target.value);
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${amountOfItems[0]} - ${amountOfItems[amountOfItems.length - 1]} of 42)`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={perPageCount}
+            onChange={event => handleChangeOption(event)}
```

Comment:
`event `- is sent by default, you don't need to specify it
```suggestion
            onChange={handleChangeOption}
```